text/0000-elide-array-size.md

```diff
@@ -0,0 +1,118 @@
+- Feature Name: elide_array_size
+- Start Date: 2018-09-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+In arrays, allow to elide the size in the type and put an underscore there
+instead if it can be deduced from the initializer. For example: `static
+BLORP_NUMBERS: [u32; _] = [0, 8, 15];`.
+
+# Motivation
+[motivation]: #motivation
+
+This will make it easier to set up static data. With the current syntax, one
+needs to count the elements of an array manually to determine its size. Letting
+the compiler find the size reduces the potential for error. It also allows us
+to ascribe the array component type without requiring the size (which is
+perhaps surprisingly not currently allowed).
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Requiring to write `[MyType; 42]` for a 42-element array gets tedious,
+especially since the compiler already knows the size from the initializer. To
+reduce the strain, Rust allows you to omit the `42` in those type ascriptions.
+
+For example, you might write:
+
+```rust
+const CONST_CHARS: [u8; _] = b"This is really a byte array";
```

Comment:
Nit: the expression here needs to be `*b"…"` (with a star added) because the type of byte string literals is `&'static [u8; N]`. (Maybe it should have been plain `[u8; N]` but oh well.)